nifi-nar-bundles/nifi-confluent-platform-bundle/nifi-confluent-schema-registry-service/src/main/java/org/apache/nifi/confluent/schemaregistry/client/RestSchemaRegistryClient.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.nifi.schema.access.SchemaNotFoundException;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.SchemaIdentifier;
+import org.apache.nifi.web.util.WebUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.node.ArrayNode;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.WebResource;
+import com.sun.jersey.api.client.config.ClientConfig;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+
+
+/**
+ * <p>
+ * A Client for interacting with Confluent Schema Registry. We make use of Jersey Client to interact with the
+ * Confluent Schema Registry REST API because the provided schema registry client does not provide a way to
+ * use HTTPS for interacting with the schema registry (it assumes that system properties will be used, instead of
+ * an SSLContext) and also does not allow configuration of (or use) timeouts. As a result, if the Schema Registry
+ * crashed or was shut down, NiFi threads could be stuck indefinitely until NiFi is restarted. To avoid this,
+ * we make use of Jersey Client and set timeouts appropriately.
+ * </p>
+ */
+public class RestSchemaRegistryClient implements SchemaRegistryClient {
+    private final List<String> baseUrls;
+    private final Client client;
+
+    private static final String SUBJECT_FIELD_NAME = "subject";
+    private static final String VERSION_FIELD_NAME = "version";
+    private static final String ID_FIELD_NAME = "id";
+    private static final String SCHEMA_TEXT_FIELD_NAME = "schema";
+
+    private final ConcurrentMap<String, Integer> schemaNameToIdentifierMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Integer, String> schemaIdentifierToNameMap = new ConcurrentHashMap<>();
+
+
+    public RestSchemaRegistryClient(final List<String> baseUrls, final int timeoutMillis, final SSLContext sslContext) {
+        this.baseUrls = new ArrayList<>(baseUrls);
+
+        final ClientConfig clientConfig = new DefaultClientConfig();
+        clientConfig.getProperties().put(ClientConfig.PROPERTY_CONNECT_TIMEOUT, timeoutMillis);
+        clientConfig.getProperties().put(ClientConfig.PROPERTY_READ_TIMEOUT, timeoutMillis);
+        client = WebUtils.createClient(clientConfig, sslContext);
+    }
+
+
+    @Override
+    public RecordSchema getSchema(final String schemaName) throws IOException, SchemaNotFoundException {
+        final String pathSuffix = getSubjectPath(schemaName);
+        final JsonNode responseJson = fetchJsonResponse(pathSuffix, "name " + schemaName);
+
+        final RecordSchema recordSchema = createRecordSchema(responseJson);
+        return recordSchema;
+    }
+
+
+    @Override
+    public RecordSchema getSchema(final int schemaId) throws IOException, SchemaNotFoundException {
+        // The Confluent Schema Registry's REST API will not provide us with the 'subject' (name) of a Schema given the ID.
```

Comment:
We should file a JIRA/Request for it. And we should change the comments to 'does not' instead of 'will not'.